app/src/command.rs

```diff
@@ -100,6 +101,17 @@ impl<'app, D: Dependencies<'app>> Application<'app, D> {
                 UnrecoverableAppFailure
             })?
             .expect("Block repository is non-empty (must always at least contain genesis)");
+        // TODO: also check block time (preferrably dynamically from a config) and
```

Comment:
It's probably worth making a new issue for this. You can include the details about how op-geth pulls txs as a background task to make sure the block time is never exceeded while building.